Code/Tools/AssetProcessor/native/resourcecompiler/rccontroller.cpp

```diff
@@ -104,8 +97,14 @@ namespace AssetProcessor
                 m_maxJobs,
                 AZStd::GetMax(m_maxJobs / 2u, 1u));
         }
-    }
 
+        // make sure that the global thread pool has enough slots to accomidate your request though, since
+        // by default, the global thread pool has idealThreadCount() slots only.
+        // leave an extra slot for non-job work.
+        int currentMaxThreadCount = QThreadPool::globalInstance()->maxThreadCount();
```

Comment:
Authors note:  I moved this section from the constructor to down here, becuase this is where it figures out what the actual max jobs are.